packages/vhd-lib/src/tests/utils.js

```diff
@@ -77,11 +75,23 @@ export async function createRandomVhdDirectory(path, sizeMB) {
   const blockDataSize = 512 * 4096
   const bitmap = Buffer.alloc(4096)
   await fs.mkdir(path + '/blocks/')
-  await fs.mkdir(path + '/blocks/1/')
+  await fs.mkdir(path + '/blocks/0/')
+  const stats = await fs.stat(rawFileName)
+
+  const sizeMB = stats.size / 1024 / 1024
   for (let i = 0, offset = 0; i < sizeMB; i++, offset += blockDataSize) {
     const blockData = Buffer.alloc(blockDataSize)
     await fs.read(srcRaw, blockData, offset)
-    await fs.writeFile(path + '/blocks/1/' + i, Buffer.concat([bitmap, blockData]))
+    await fs.writeFile(path + '/blocks/0/' + i, Buffer.concat([bitmap, blockData]))
```

Comment:
Be careful, this does not look like a valid bitmap, it appears to be all empty.

Reply:
bitmap is never used in xo , since we're working at the block level

Reply:
Won't it be used when creating a stream (or a file) from this VHD?

Either:
- include a correct bitmap in this block files
- don't include the bitmap and create one when necessary.

Reply:
ok, I built a full bitmap